cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -452,7 +455,8 @@ private void logProgramStart(Program program, ProgramOptions options) {
     Cancellable saveContextCancellable =
       LoggingContextAccessor.setLoggingContext(loggingContext);
     String userArguments = Joiner.on(", ").withKeyValueSeparator("=").join(options.getUserArguments());
-    LOG.info("Starting {} Program '{}' with Arguments [{}]", program.getType(), program.getName(), userArguments);
+    LOG.info("Starting {} Program '{}' with Arguments [{}]. Executing user is {}",
+             program.getType(), program.getName(), userArguments, authenticationContext.getPrincipal().getName());
```

Comment:
Instead of `authenticationContext.getPrincipal().getName()`, isn't it possible to simply log `UserGroupInformation.getCurrentUser().getUsername()`?

Reply:
This will log the impersonated user right? But I think we want to log the user which actually starts the program

Reply:
Do we have a JIRA for this? What is this for?

Reply:
One of our customer wants to know which logged in user starts the program in the log file. That's the motivation of this pr. Created JIRA: https://issues.cask.co/browse/CDAP-13150

Reply:
This might not work as intended. Depending on how the program is launched (via REST vs via scheduler), It can give you different result

Reply:
If it is via REST, the user is in logged. If it is the scheduler, the user will be CDAP I believe

Reply:
If it is not via REST, I believe it will log the impersonated app owner. Is this the desired behavior?